woocommerce/woocommerce-android
# Update navigation library

This update is blocked by the pending PR https://github.com/woocommerce/woocommerce-android/pull/5755
### Description
The `beta` version of navigation had a bug that prevented restoring custom Parcelables from the saved state, which resulted in using a [workaround](https://github.com/woocommerce/woocommerce-android/pull/5307) to prevent the crash.
Now, Google has fixed the [bug](https://issuetracker.google.com/issues/207315994) we reported in the new RC version, this PR updates the navigation library, and removes the previous workaround.

### Testing instructions
Confirm the previous bug is fixed:

1. Enable "Don't keep activities" from the developer options.
2. Open Product Images screen of on the products.
3. Navigate to another app, then re-open the app.
4. Confirm that the app doesn't crash.
